concordium-consensus/src/Concordium/GlobalState/Persistent/Account.hs

```diff
@@ -1,928 +1,544 @@
 -- of the same account.
-serializeAccount :: forall m av. (MonadBlobStore m, MonadPut m, IsAccountVersion av) => GlobalContext -> PersistentAccount av -> m ()
-serializeAccount cryptoParams PersistentAccount{..} = do
-    PersistingAccountData {..} <- refLoad _persistingData
-    let
-        initialCredId = credId (Map.findWithDefault
-                (error "Account missing initial credential")
-                initialCredentialIndex
-                _accountCredentials
-              )
-        asfExplicitAddress = _accountAddress /= addressFromRegIdRaw initialCredId
-        -- There is an opportunity for improvement here. There is no need to go
-        -- through the deserialized key. The way the encryption key is formed is
-        -- that the first half is the generator, the second half is the credId.
-        -- So we could just concatenate them. This requires a bit of scaffolding
-        -- to get the right component out of cryptoParams, so it is not yet
-        -- done.
-        asfExplicitEncryptionKey = _accountEncryptionKey /= toRawEncryptionKey (makeEncryptionKey cryptoParams (unsafeCredIdFromRaw initialCredId))
-        (asfMultipleCredentials, putCredentials) = case Map.toList _accountCredentials of
-          [(i, cred)] | i == initialCredentialIndex -> (False, put cred)
-          _ -> (True, putSafeMapOf put put _accountCredentials)
-        asfThresholdIsOne = aiThreshold _accountVerificationKeys == 1
-        asfHasRemovedCredentials = _accountRemovedCredentials ^. unhashed /= EmptyRemovedCredentials
-    aea <- refLoad _accountEncryptedAmount
-    (asfExplicitEncryptedAmount, putEA) <- putAccountEncryptedAmountV0 aea <&> \case
-        Nothing -> (False, return ())
-        Just p -> (True, p)
-    arSched <- loadPersistentAccountReleaseSchedule =<< refLoad _accountReleaseSchedule
-    let
-        asfExplicitReleaseSchedule = arSched /= Transient.emptyAccountReleaseSchedule
-        asfHasBakerOrDelegation = case _accountStake of
-          PersistentAccountStakeNone -> False
-          _ -> True
-    stake <- loadAccountStake _accountStake
-    liftPut $ do
-        put AccountSerializationFlags {..}
-        when asfExplicitAddress $ put _accountAddress
-        when asfExplicitEncryptionKey $ put _accountEncryptionKey
-        unless asfThresholdIsOne $ put (aiThreshold _accountVerificationKeys)
-        putCredentials
-        when asfHasRemovedCredentials $ put (_accountRemovedCredentials ^. unhashed)
-        put _accountNonce
-        put _accountAmount
-        putEA
-        when asfExplicitReleaseSchedule $ put arSched
-        when asfHasBakerOrDelegation $ serializeAccountStake stake
+serializeAccount :: (MonadBlobStore m, MonadPut m) => GlobalContext -> PersistentAccount av -> m ()
+serializeAccount gc (PAV0 acc) = V0.serializeAccount gc acc
+serializeAccount gc (PAV1 acc) = V0.serializeAccount gc acc
+serializeAccount gc (PAV2 acc) = V1.serializeAccount gc acc
```

Comment:
I am not removing it since it is still used for migration in the basic block state, and it is simpler to retain it than remove everything that is involved at present.